src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -890,6 +946,74 @@ MulticopterPositionControl::failsafe(vehicle_local_position_setpoint_s &setpoint
 	}
 }
 
+void
+MulticopterPositionControl::update_avoidance_waypoint_desired(PositionControlStates &states)
+{
+	_traj_wp_avoidance_desired = _flight_tasks.getAvoidanceWaypoint();
+
+	_traj_wp_avoidance_desired.timestamp = hrt_absolute_time();
+	_traj_wp_avoidance_desired.type = vehicle_trajectory_waypoint_s::MAV_TRAJECTORY_REPRESENTATION_WAYPOINTS;
+
+	states.position.copyTo(_traj_wp_avoidance_desired.waypoints[vehicle_trajectory_waypoint_s::POINT_0].position);
+	states.position.copyTo(_traj_wp_avoidance_desired.waypoints[vehicle_trajectory_waypoint_s::POINT_0].velocity);
+	states.position.copyTo(_traj_wp_avoidance_desired.waypoints[vehicle_trajectory_waypoint_s::POINT_0].acceleration);
+	_traj_wp_avoidance_desired.waypoints[vehicle_trajectory_waypoint_s::POINT_0].yaw = states.yaw;
+	_traj_wp_avoidance_desired.waypoints[vehicle_trajectory_waypoint_s::POINT_0].yaw_speed = NAN;
```

Comment:
If the yawspeed is something you need in the future we could add it to the available states...